virtualizarr/writers/icechunk.py

```diff
@@ -124,15 +134,37 @@ def write_virtual_variable_to_icechunk(
     group: "Group",
     name: str,
     var: Variable,
+    append_dim: Optional[str] = None,
 ) -> None:
     """Write a single virtual variable into an icechunk store"""
     ma = cast(ManifestArray, var.data)
     zarray = ma.zarray
+    shape = zarray.shape
+    mode = store.mode.str
+
+    # Aimee: resize the array if it already exists
+    # TODO: assert chunking and encoding is the same
+    existing_keys = tuple(group.array_keys())
+    append_axis, existing_num_chunks = None, None
+    if name in existing_keys and mode == "a":
+        # resize
+        dims = var.dims
+        if append_dim in dims:
+            append_axis = dims.index(append_dim)
+            existing_array = group[name]
+            existing_size = existing_array.shape[append_axis]
+            existing_num_chunks = int(
+                existing_size / existing_array.chunks[append_axis]
+            )
```

Comment:
There's a whole beartrap here around noticing if the last chunk is smaller than the other chunks. We should throw in that case (because zarr can't support it without variable-length chunks).